crates/sui-core/src/authority_active/gossip/mod.rs

```diff
@@ -261,7 +267,10 @@ where
 
                 items = &mut streamx.next() => {
                     match items {
-                        Some(Ok(BatchInfoResponseItem(UpdateItem::Batch(_signed_batch)) )) => {},
+                        Some(Ok(BatchInfoResponseItem(UpdateItem::Batch(signed_batch)) )) => {
+                            let next_seq = signed_batch.batch.next_sequence_number;
+                            self.state.record_next_sequence(self.peer_name, next_seq)?;
+                        },
```

Comment:
We see a batch before we see the transactions for the batch, so even though a batch header arrives, it does not guarantee that the transactions will be processed. The `self.state.record_next_sequence(self.peer_name, next_seq)?;` should happen in the case for receiving a transaction below, or this can also record the start sequence number of the batch instead of the next sequence since the start of this batch means that the previous batch's transactions were already processed.

Reply:
Correction, the sequence is initial batch, transactions batch that signs proceeding transactions. No code change is necessary here.